src/tensor_ops/realize_to.rs

```diff
@@ -0,0 +1,214 @@
+use crate::{shapes::*, tensor::*};
+
+/// Changes order of dimensions/axes
+pub trait RealizeTo: HasErr + HasShape {
+    /// Realizes the concrete shape of the tensor as another compatable shape,
+    /// or returns the original tensor if the new shape's dimensions are incompatable.
+    /// ```rust
+    /// # use dfdx::prelude::*;
+    /// # let dev: Cpu = Default::default();
+    /// let a: Tensor<Rank2<2, 3>, f32, _> = dev.zeros();
+    /// let a = a.realize::<(usize, usize)>().unwrap();
+    /// let mut a = a.realize::<Rank2<2, 3>>().unwrap();
+    /// match a.realize::<(usize, Const<4>)>() {
+    ///     Ok(new) => handle_realized(new),
+    ///     Err(old) => handle_failure(old),
```

Comment:
For the sake of simple example we can just do a print statement:
```suggestion
    ///     Ok(new) => println!("Shape was properly realized, returned new tensor"),
    ///     Err(old) => println!("Shape could not be realized, returned the original tensor"),
```

Reply:
Thanks!